hubblestack/extmods/returners/sqlite.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+'''
+Return hubble data to ssqlite (intended for testing)
+'''
+from __future__ import absolute_import
+
+import salt.returners
+import logging
+import os
+
+__virtualname__ = 'sqlite'
+
+log = logging.getLogger(__virtualname__)
+
+_conn = None
+
+def __virtual__():
+    '''
+    Return virtual name of the module.
+
+    :return: The virtual name of the module.
+    '''
+    global json
+    global sqlite3
+    try:
+        import json
+        GOT_JSON = True
+    except:
+        pass
+    try:
+        import sqlite3
+        GOT_SQLI = True
+    except:
+        pass
+    if GOT_JSON and GOT_SQLI:
+        return __virtualname__
+    msgs = []
+    if not GOT_JSON:
+        msgs.append("json module is missing")
+    if not GOT_SQLI:
+        msgs.append("sqlite3 module is missing")
+    return False, ', '.join(msgs)
+
+def _get_options(ret=None):
+    '''
+    Get the sqlite dumpster options from configs
+
+    :return: options
+    '''
+
+    defaults = {'dumpster': '/var/log/hubblestack/returns.sqlite'}
+
+    attrs = {'dumpster': 'dumpster'}
+
+    _options = salt.returners.get_returner_options(__virtualname__,
+                                                   ret,
+                                                   attrs,
+                                                   __salt__=__salt__,
+                                                   __opts__=__opts__,
+                                                   defaults=defaults)
+    log.debug("_options: {}".format(_options))
+    return _options
+
+def _get_conn():
+    '''
+    Establish a connection (if not connected) and return it
+
+    :return: connection
+    '''
+    _options = _get_options()
+    global _conn
+    if not _conn:
+        _p = _options.get('dumpster','hubble-returns.db')
+        _d = os.path.dirname(_p)
+        if _d and not os.path.isdir(_d):
+            log.debug('creating directory {0}'.format(_d))
+            os.makedirs(_d, 0755)
+        log.debug('connecting to database in {0}'.format(_p))
+        _conn = sqlite3.connect( _options.get('dumpster', 'hubble-returns.db') )
+        log.debug('creating ret table')
+        _conn.execute('''create table if not exists ret(
+            jid text, id text, fun text, fun_args json,
+            ret json)''')
+    return _conn
+
+## {u'fun': u'hubble.audit',
```

Comment:
Was this just a test event you were referencing while you were building out the returner?

Reply:
Yeah, it's for reference. Should I remove that?